### PR TITLE
Device settings endpoints tests.

### PR DESCRIPTION
**Pull request by ice-myles**
Device settings endpoints tests: CreateDeviceSettings, ModifyDeviceSettings and GetDeviceSettings.

### CODE REVIEW

File: cmd/eskimo/devices_test.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+func TestEskimo_GetUserDevices_Failure_OperationNotAllowed(t *testing.T) {
+	t.Parallel()
+	if testing.Short() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), testDeadline)
+	t.Cleanup(cancel)
+
+	// We launch scenario N times in parallel to check functionality with the different random data not to be lucky with one data set only.
+	repeatScenarioInParallel(t, func() {
```
Comment (victor-oancea-leftclick-io) [inline cmd/eskimo/devices_test.go:129]: No, we dont do that.

Remove this completely. Bad design. Tests have to be as deterministic as posible

File: users/.testdata/application.yaml [hunk 2]
```diff
@@ -47,6 +47,7 @@ users: &users
     consumingTopics:
       - users-events
       - mining-started-events
+      - user-device-settings-events
```
Comment (victor-oancea-leftclick-io) [inline users/.testdata/application.yaml:50]: Nope, think again

File: cmd/fixture/fixture.go [hunk 3]
```diff
@@ -311,6 +314,44 @@ func (b *TestConnectorsBridge) MustCreateDefaultUser(ctx context.Context, tb tes
...
+	_, reqBodyDefault := b.MustGenerateDeviceSettingsReqBodies(tb, userID, deviceID)
+	body, status := b.CreateDeviceSettings(ctx, tb, userID, deviceID, token, reqBodyDefault)
+	assert.Equal(tb, 201, status) //nolint:gomnd // Nothing magical about it.
+
+	var expected users.DeviceSettings
+	require.NoError(tb, json.Unmarshal([]byte(reqBodyDefault), &expected))
+	b.AssertCreatedOrModifiedDeviceSettingsResponseBody(tb, &expected, body)
+}
+
+func (*TestConnectorsBridge) MustGenerateDeviceSettingsReqBodies(
```
Comment (nusanovkornilov-leftclick) [inline cmd/fixture/fixture.go:328]: Why not to split with/without notification settings to different functions? As I can see, 90% of time u're using only one of them

File: cmd/fixture/fixture.go [hunk 4]
```diff
@@ -404,3 +453,14 @@ func (*TestConnectorsBridge) AssertResponseBody(tb testing.TB, expectedRespBody,
 	tb.Helper()
 	assert.Regexp(tb, regexp.MustCompile(strings.ReplaceAll(strings.ReplaceAll(expectedRespBody, "\t", ""), "\n", "")), actualRespBody)
 }
+
+func (*TestConnectorsBridge) RandomHex(t *testing.T, num int) string {
```
Comment (nusanovkornilov-leftclick) [inline cmd/fixture/fixture.go:457]: Unused? Anyway you can use uuid.NewString(), do you really need variabl length?

**Review by nusanovkornilov-leftclick — CHANGES_REQUESTED** [review]

**Pull request closed without merging**